server/src/main/java/org/opensearch/index/store/DirectoryFileTransferTracker.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.store;
+
+import org.opensearch.common.util.MovingAverage;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+
+import java.io.IOException;
+
+/**
+ * Tracks the amount of bytes transferred between two {@link org.apache.lucene.store.Directory} instances
+ *
+ * @opensearch.internal
+ */
+public class DirectoryFileTransferTracker {
+    /**
+     * Cumulative size of files (in bytes) attempted to be copied over from the source {@link org.apache.lucene.store.Directory}
+     */
+    private volatile long downloadBytesStarted;
```

Comment:
We may need to rename `download` to something else. This tracker does not necessarily know if the files are copied (and hence download) from remote directory. It can very well be some other directory.